common/logging.go

```diff
@@ -0,0 +1,266 @@
+// Copyright 2016-2017 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package common
+
+import (
+	"fmt"
+	"io/ioutil"
+	"log/syslog"
+	"net"
+	"os"
+	"strconv"
+	"time"
+
+	"github.com/Sirupsen/logrus"
+	"github.com/Sirupsen/logrus/hooks/syslog"
+	"github.com/bshuster-repo/logrus-logstash-hook"
+	"github.com/evalphobia/logrus_fluent"
+	"regexp"
+)
+
+// syslogOpts is the set of supported options for syslog configuration.
+var syslogOpts = map[string]bool{
+	"syslog.level": true,
+}
+
+// fluentDOpts is the set of supported options for fluentD configuration.
+var fluentDOpts = map[string]bool{
+	"fluentd.address": true,
+	"fluentd.tag":     true,
+	"fluentd.level":   true,
```

Comment:
Would be great to move these into constants as we rely on the same string value in multiple places.